### PR TITLE
initial attempt at  github actions workflow for formatting and testing

### PR DESCRIPTION
**Pull request by Eric-Butcher**

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install poetry
+        poetry install
+
+    - name: Sort imports with isort
+      run: poetry run isort . 
+
+    - name: Format with autopep8
+      run: poetry run autopep8 .
```
Comment (WeirdAlex03) [inline .github/workflows/formatting.yml]: You need to specify the `-ri` flags to run through the directory `r`ecursively and fix the problems `i`n-place (as opposed to just logging problems to file)

File: .github/workflows/formatting.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      run: poetry run isort . 
+
+    - name: Format with autopep8
+      run: poetry run autopep8 . 
+
+    - name: Lint with flake8
+      run: poetry run flake8 .
+
+    - name: Find problems from mypy
+      run: poetry run mypy .
```
Comment (WeirdAlex03) [inline .github/workflows/formatting.yml]: This gets a bunch of `import-untyped` problems from Django. Given that's unavoidable on this project, adding `--disable-error-code import-untyped` will mute those

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        python-version: 3.12
+    - name: Install environment
+      run: |
+        python -m pip install --upgrade pip
+        pip install poetry
+        poetry install
+
+    - name: Django Testing
+      run: |
+        poetry run manage.py test
```
Comment (WeirdAlex03) [inline .github/workflows/testing.yml]: This should be `poetry run ./manage.py test`, though that now fails for a different reason for me. Looks like a Django config problem now I think? Should compare the settings.py as of #12 with current to see what's different

File: evergreen/pyproject.toml [hunk 4]
```diff
@@ -23,6 +23,10 @@ psycopg = { extras = ["binary"], version = "^3.2.3" }
 
 Pillow = "^9.2.0"
 django-bootstrap = "^0.2.4"
+flake8 = "^7.1.1"
+autopep8 = "^2.3.1"
+isort = "^5.13.2"
+mypy = "^1.13.0"
```
Comment (WeirdAlex03) [inline evergreen/pyproject.toml]: Does Poetry happen to support dev-envorinment-only packages like npm's `--save-dev`? Obviously we don't need linting/formatting in prod, could save some time on deployment waiting for the Docker container to come online

File: .github/workflows/formatting.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
```
Comment (WeirdAlex03) [inline .github/workflows/formatting.yml]: At least `isort` and `autopep8` will make changes directly to the files. You should add the commit action to save them when done

**Review by WeirdAlex03 — COMMENTED** [review]
I know you're not done yet, but added some feedback from another pair of eyes anyway

File: evergreen/pyproject.toml (hunk 4 above)
Comment (Eric-Butcher) [reply evergreen/pyproject.toml]: This is technically true but I'd rather not deal with it right now.

File: .github/workflows/formatting.yml (hunk 5 above)
Comment (Eric-Butcher) [reply .github/workflows/formatting.yml]: Yea I forgot about this but also don't want to be committing stuff until I know everything works right

File: .github/workflows/testing.yml (hunk 3 above)
Comment (Eric-Butcher) [reply .github/workflows/testing.yml]: Yes the test for manage.py keeps failing for me and Im not quite sure why. The errors are in the manage.py code so.....

**Review by WeirdAlex03 — APPROVED** [review]

**Pull request merged**